src/api/app/components/workflow_artifacts_per_step_component.rb

```diff
@@ -0,0 +1,90 @@
+class WorkflowArtifactsPerStepComponent < ApplicationComponent
+  with_collection_parameter :artifacts_per_step
+
+  attr_reader :artifacts_per_step, :step, :artifacts
+
+  def initialize(artifacts_per_step:)
+    super
+    @artifacts_per_step = artifacts_per_step
+    @step = @artifacts_per_step.step
+    @artifacts = @artifacts_per_step.artifacts
+  end
+
+  def artifacts_sentence(context)
+    parsed_artifacts = JSON.parse(artifacts).deep_symbolize_keys
+
+    case step
+    when 'Workflow::Step::BranchPackageStep'
+      artifacts_for_branch_or_link_package(context, parsed_artifacts, 'Branched package from ')
+    when 'Workflow::Step::LinkPackageStep'
+      artifacts_for_branch_or_link_package(context, parsed_artifacts, 'Linked package from ')
+    when 'Workflow::Step::RebuildPackage'
+      artifacts_for_rebuild_package(context, parsed_artifacts)
+    when 'Workflow::Step::ConfigureRepositories'
+      artifacts_for_configure_repositories(context, parsed_artifacts)
+    end
+  rescue JSON::ParserError
+    nil
```

Comment:
We should display at least an error, otherwise users won't know what happened.

Reply:
done